core/services/job/fetchers.go

```diff
@@ -0,0 +1,147 @@
+package job
+
+import (
+	"encoding/json"
+	"sort"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+type Fetchers []Fetcher
+
+func (f *Fetchers) UnmarshalJSON(bs []byte) error {
+	var spec struct {
+		Fetchers []json.RawMessage `json:"fetchers"`
+	}
+	err := json.Unmarshal(bs, &spec)
+	if err != nil {
+		return err
+	}
+
+	for _, fetcherBytes := range spec.Fetchers {
+		var fetcherSpec struct {
+			Type FetcherType `json:"type"`
+		}
+		err := json.Unmarshal([]byte(fetcherBytes), &fetcherSpec)
+		if err != nil {
+			return err
+		}
+
+		var fetcher Fetcher
+		switch fetcherSpec.Type {
+		case FetcherTypeBridge:
+			fetcher = BridgeFetcher{}
+		case FetcherTypeHttp:
+			fetcher = HttpFetcher{}
+		case FetcherTypeMedian:
+			fetcher = MedianFetcher{}
+		default:
+			return errors.New("unknown fetcher type")
+		}
+
+		err = json.Unmarshal([]byte(fetcherBytes), &fetcher)
+		if err != nil {
+			return err
+		}
+
+		j.Fetchers = append(j.Fetchers, fetcher)
```

Comment:
I dont know if its more idiomatic to assign length to a slice at the beginning and then allocate with index?